app/src/main/java/com/infomaniak/drive/data/services/UploadWorker.kt

```diff
@@ -325,6 +315,18 @@ class UploadWorker(appContext: Context, params: WorkerParameters) : CoroutineWor
         jobs.joinAll()
     }
 
+    private fun getImagesConditions(): String = when {
+        Build.VERSION.SDK_INT > Build.VERSION_CODES.R -> "AND ${MediaStore.Images.Media.IS_PENDING} = 0 AND ${MediaStore.Images.Media.IS_TRASHED} = 0"
+        Build.VERSION.SDK_INT > Build.VERSION_CODES.Q -> "AND ${MediaStore.Images.Media.IS_PENDING} = 0"
+        else -> ""
+    }
+
+    private fun getVideosConditions(): String = when {
+        Build.VERSION.SDK_INT > Build.VERSION_CODES.R -> "AND ${MediaStore.Video.Media.IS_PENDING} = 0 AND ${MediaStore.Video.Media.IS_TRASHED} = 0"
+        Build.VERSION.SDK_INT > Build.VERSION_CODES.Q -> "AND ${MediaStore.Video.Media.IS_PENDING} = 0"
+        else -> ""
+    }
```

Comment:
use a single fun with `MediaStore.MediaColumns.IS_PENDING` and `MediaStore.MediaColumns.IS_TRASHED`